python/tskit/tables.py

```diff
@@ -2601,6 +2601,24 @@ def sort(self, edge_start=0):
         self._ll_tables.sort(edge_start)
         # TODO add provenance
 
+    def canonicalise(self):
```

Comment:
I think we should add ``filter_populations=True, filter_individuals=True`` here just like in simplify.  Shouldn't the default behaviour of canonicalise be to remove unreferenced entities? That way, if you call canonicalise on two things that slightly different versions of redundant information you'll get the same output, which, I would think, is what you want.